### PR TITLE
Release 23.2

### PR DESCRIPTION
**Pull request by blackboxsw**
Bump the version in cloudinit/version.py to 23.2 and update ChangeLog.

## Proposed Commit Message

```
Release 23.2

Bump the version in cloudinit/version.py to 23.2 and
update ChangeLog.
```

## Additional Context

https://github.com/orgs/canonical/projects/29/views/1?pane=issue&itemId=21361535
## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/contributing.html)
 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
I should have done this earlier, but let's include 23.1.2 in the ChangeLog:
```
23.1.2
 - Make user/vendor data sensitive and remove log permissions
   (LP: #2013967) (CVE-2023-1786)
```

Otherwise, LGTM

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,155 @@
...
+ - setup.py: use pkg-config for udev/rules path (#2137) [dankm]
+ - openstack/static: honor the DNS servers associated with a network
+   (#2138) [Gonéri Le Bouder]
+ - fix linux-specific code on bsd (#2143)
+ - cli: schema validation of jinja template user-data (SC-1385) (#2132)
+   (LP: #1881925)
+ - gce: activate network discovery on every boot (#2128)
+ - tests: update integration test to assert 640 across reboots (#2145)
+ - Make user/vendor data sensitive and remove log permissions (#2144)
+   (LP: #2013967)
```
Comment (blackboxsw) [inline ChangeLog:65]: @TheRealFalcon This content is officially released already as 23.1.2. shall we redact it

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

File: ChangeLog (hunk 1 above)
Comment (blackboxsw) [reply ChangeLog:65]: Opted to leave duplicated content in "security releases" given that most reviewers of 23.2 will be looking at wholesale set of changes between 23.2 and 23.1... and having the hot-fixes included in 23.2 section makes sense.

**Pull request merged**